homeassistant/components/device_tracker/ubus.py

```diff
@@ -107,54 +112,70 @@ def _update_info(self):
             _LOGGER.info("Checking ARP")
 
             if not self.hostapd:
-                hostapd = _req_json_rpc(
-                    self.url, self.session_id, 'list', 'hostapd.*', '')
+                hostapd = self._req_json_rpc('list', 'hostapd.*', '')
                 self.hostapd.extend(hostapd.keys())
 
             self.last_results = []
             results = 0
             for hostapd in self.hostapd:
-                result = _req_json_rpc(
-                    self.url, self.session_id, 'call', hostapd, 'get_clients')
+                result = self._req_json_rpc('call', hostapd, 'get_clients')
 
                 if result:
                     results = results + 1
                     self.last_results.extend(result['clients'].keys())
 
             return bool(results)
 
+    def _req_json_rpc(self, rpcmethod, subsystem, method, **params):
+        """Perform one JSON RPC operation."""
+        retry_count = 2
+        while retry_count >= 0:
+            retry_count -= 1
+            data = json.dumps({"jsonrpc": "2.0",
+                               "id": 1,
+                               "method": rpcmethod,
+                               "params": [self.session_id,
+                                          subsystem,
+                                          method,
+                                          params]})
 
-def _req_json_rpc(url, session_id, rpcmethod, subsystem, method, **params):
-    """Perform one JSON RPC operation."""
-    data = json.dumps({"jsonrpc": "2.0",
-                       "id": 1,
-                       "method": rpcmethod,
-                       "params": [session_id,
-                                  subsystem,
-                                  method,
-                                  params]})
-
-    try:
-        res = requests.post(url, data=data, timeout=5)
-
-    except requests.exceptions.Timeout:
-        return
-
-    if res.status_code == 200:
-        response = res.json()
-
-        if rpcmethod == "call":
             try:
-                return response["result"][1]
-            except IndexError:
-                return
-        else:
-            return response["result"]
+                res = requests.post(self.url, data=data, timeout=5)
 
+            except requests.exceptions.Timeout:
+                return
 
-def _get_session_id(url, username, password):
-    """Get the authentication token for the given host+username+password."""
-    res = _req_json_rpc(url, "00000000000000000000000000000000", 'call',
-                        'session', 'login', username=username,
-                        password=password)
-    return res["ubus_rpc_session"]
+            if res.status_code == 200:
```

Comment:
To keep the code cleaner, use a guard clause to return as soon as you know what you're going to return.

```python
if res.status_code != 200:
    return

response = res.json()
```